### ISSUE TITLE
Introduce wollok.lang.Closure

### ISSUE THREAD

**Issue by javierfernandes** [uqbar-project/wollok#369]
I believe that this requires either having many classes like Closure0, Closure1, Closure2, Closure3.. .etc.. for different numbers of parameters, either supporting varargs for the "apply method"

```
method apply(arguments...) {
}

```

**Comment by npasserini** [comment]
I like the varargs!

**Comment by javierfernandes** [comment]
Varargs already implemented in #371 
Now we just need to model the wollok.lang.Closure and remove the 100% native class and refactor the whole interpreter

**Comment by javierfernandes** [comment]
Done !

For the moment, it just has this

```xtend
	/**
	 * @author jfernandes
	 * @since 1.3
	 * @noInstantiate
	 */
	class Closure {
		method apply(parameters...) native
		method toString() native
	}
```
